examples/notebooks/get-started/learn_beam_transforms_by_doing.ipynb

```diff
@@ -0,0 +1,750 @@
+        "# EDIT THIS CODE CELL\n",
+        "\n",
+        "inputs_pattern = 'data/*'\n",
+        "outputs_prefix = 'outputs/part3'\n",
+        "\n",
+        "# Running locally in the DirectRunner.\n",
+        "with beam.Pipeline() as pipeline:\n",
+        "  word_count = (\n",
+        "      pipeline\n",
+        "        | 'Read lines' >> beam.io.ReadFromText(inputs_pattern)\n",
+        "        | 'Find words' >> beam.FlatMap(lambda line: re.findall(r\"[a-zA-Z']+\", line))\n",
+        "        | 'Count words' >> beam.combiners.Count.PerElement()\n",
+        "        | 'Write results' >> beam.io.WriteToText(outputs_prefix)\n",
+        "  )\n",
+        "\n",
+        "pipeline.run()\n",
+        "# Sample the first 20 results, remember there are no ordering guarantees.\n",
+        "run('head -n 20 {}-00000-of-*'.format(outputs_prefix))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "cellView": "form",
+        "id": "6Cn4pk3j8K4p"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Answer\n",
+        "inputs_pattern = 'data/*'\n",
+        "outputs_prefix = 'outputs/part3'\n",
+        "\n",
+        "# Running locally in the DirectRunner.\n",
+        "with beam.Pipeline() as pipeline:\n",
+        "  word_count = (\n",
+        "      pipeline\n",
+        "        | 'Read lines' >> beam.io.ReadFromText(inputs_pattern)\n",
+        "        # The composite transform function you created above\n",
+        "        | 'Count Words' >> CountWords()\n",
+        "        | 'Write results' >> beam.io.WriteToText(outputs_prefix)\n",
+        "  )\n",
+        "\n",
+        "# Sample the first 20 results, remember there are no ordering guarantees.\n",
+        "run('head -n 20 {}-00000-of-*'.format(outputs_prefix))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "w73aU2IsDAS1"
+      },
+      "source": [
+        "### 3.3 Creating Your Own Composite Transform With `PTransform` Directly"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "uLEAICZKDz6e"
+      },
+      "source": [
+        "To create your own composite transform, create a subclass of the `PTransform` class and override the `expand` method to specify the actual processing logic\n",
+        "([more details](https://beam.apache.org/documentation/programming-guide/#composite-transform-creation))."
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JT_ZtzjIEHhD"
+      },
+      "source": [
+        "For example, if we wanted to create our own composite transform that counted the length of each word.\n",
+        "\n",
+        "The following code sample shows how to declare a `PTransform` that accepts a `PCollection` of Strings for input, and outputs a `PCollection` of integers\n",
```

Comment:
Can we link the first code cell directly below this?

Reply:
Good point. Moved the cell.